pennylane/templates/swap_networks/ccl2.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the TwoLocalSwapNetwork template.
+"""
+
+import warnings
+import pennylane as qml
+import pennylane.numpy as np
+from pennylane.operation import Operation, AnyWires
+from pennylane.ops import SWAP, FermionicSWAP
+
+
+class TwoLocalSwapNetwork(Operation):
+    r"""Apply two-local gate operations using a canonical 2-complete linear (2-CCL) swap network.
+
+    Args:
+        wires (Iterable or Wires): ordered sequence of wires on which the swap network acts
+        acquaintances (Callable): a callable `func(index, wires, param=None, **kwargs)` that returns a two-local operation being applied on pair of wires specified by `index` that are currently stored in physical qubits or fermionic modes specified by `wires` before they are swapped apart and kwargs accepts any additional keyword argument required for this operation. Note that these kwargs should be provided separately
+        weights (tensor): one dimensional tensor to provide weights for the acquaintances of length :math:`N \times (N - 1) / 2`, where `N` is the number of wires and each weight is specified by the `param` argument in the callable given for `acquaintances`. Specify ``None`` if specified acquintances will not be parameterized
+        fermionic (bool): If ``True``, qubits are realized as fermionic modes and :class:`~.pennylane.FermionicSWAP` with :math:`\phi=2\pi` is used instead of class:`~.pennylane.SWAP` to build 2-CCL
+        shift (bool): If ``True``, odd-numbered layers begins from the second qubit instead of first
```

Comment:
I don't have a good suggestion right now, but there is a lot of text in these argument descriptions. Is it possible to make it more concise without losing the important details?

Reply:
I have tried to make them crisper, let me know if they require more work.